packages/modelviewer.dev/src/docs-and-examples/create-html.ts

```diff
@@ -0,0 +1,238 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+interface Entry {
+  name: string, htmlName: string, description: string, default: string[],
+      links: string[],
+}
+
+interface Category {
+  Title: string, Attributes: Entry[], 'CSS Custom Properties': Entry[],
+      Properties: Entry[], Methods: Entry[], Events: Entry[], Slots: Entry[],
+}
+
+const CategoryConstant: Category = {
+  Title: '',
+  Attributes: [],
+  'CSS Custom Properties': [],
+  Properties: [],
+  Methods: [],
+  Events: [],
+  Slots: [],
+}
+
+function createSubcategorySidebar(subcategory: string, lowerCaseTitle: string):
+    string {
+      const lowerCaseKey = getLowerCaseKey(subcategory);
+      return `
+    <div class="subCategory" id=${'subCategory'.concat(subcategory)}>
+      <h4 class="subcategory-header" id=${
+          lowerCaseTitle.concat('-', lowerCaseKey, '-sidebar')}>
+        <a class="darken" href="#${lowerCaseTitle.concat('-', lowerCaseKey)}">${
+          subcategory}</a>
+      <h4>
+    </div>
+  `;
+    }
+
+function createSidebar(category: Category, docsOrExamples: 'docs'|'examples') {
+  if (docsOrExamples === 'examples')
+    return;
+  const container = document.getElementById('sidebar-category-container');
+  const lowerCaseTitle = category.Title.toLowerCase();
+  let subcategories = Object.keys(category);
+  subcategories = subcategories.filter(k => k !== 'Title');
+
+  const categoryContainer = `
+    <div class="category" id=${lowerCaseTitle.concat('aboveHeader')}>
+      <h3 id=${lowerCaseTitle.concat('-sidebar')}>
+        <a class="darken" href="#${lowerCaseTitle}">${category.Title}</a>
+      <h3>
+    </div>
+  `;
+
+  container!.innerHTML += categoryContainer;
+
+  const innerCategory =
+      document.getElementById(lowerCaseTitle.concat('aboveHeader'));
+  for (const subcategory of subcategories) {
+    innerCategory!.innerHTML +=
+        createSubcategorySidebar(subcategory, lowerCaseTitle);
+
+    const innerSubcategory =
+        document.getElementById(lowerCaseTitle.concat('aboveHeader'));
+    const lowerCaseKey = getLowerCaseKey(subcategory);
+    const entries =
+        (<Entry[]>category[subcategory as keyof typeof CategoryConstant]);
+    for (const entry of entries) {
+      innerSubcategory!.innerHTML += `
+        <div class="element de-active" id=${
+          lowerCaseTitle.concat(
+              '-', lowerCaseKey, '-', entry.htmlName)}>
+          <a class="darken" href=${
+                                                     '#docs-'.concat(
+                                                         lowerCaseTitle,
+                                                         '-',
+                                                         lowerCaseKey,
+                                                         '-',
+                                                         entry.htmlName)}>${
```

Comment:
Whenever auto-formatting gives you horrible looking results like this, it's a good indicator that there's a better way. I'd recommend giving some of these `concat()`s names and using those here instead.

Reply:
done